kura/org.eclipse.kura.net.admin/src/main/java/org/eclipse/kura/net/admin/NetworkAdminServiceImpl.java

```diff
@@ -551,189 +548,185 @@ public void updateWifiInterfaceConfig(String interfaceName, boolean autoConnect,
-                                        newNetConfigs.add(netConfig);
-                                        configurationChanged = true;
-                                        if (!modifiedInterfaceNames.contains(interfaceName)) {
-                                            modifiedInterfaceNames.add(interfaceName);
-                                        }
+                            } else if (netConfig instanceof DhcpServerConfigIP4) {
+                                if (dhcpServerConfigIP4 == null) {
+                                    logger.debug("removing DhcpServerConfigIP4 for {}", interfaceName);
+                                    configurationChanged = true;
+                                    if (!modifiedInterfaceNames.contains(interfaceName)) {
+                                        modifiedInterfaceNames.add(interfaceName);
                                     }
-                                    if (netConfig instanceof NetConfigIP6 && !hadNetConfig6) {
-                                        logger.debug("adding new NetConfig6 to existing config for {}", interfaceName);
-                                        newNetConfigs.add(netConfig);
+                                } else {
+                                    hadDhcpServerConfigIP4 = true;
+                                    newNetConfigs.add(dhcpServerConfigIP4);
+                                    if (!netConfig.equals(dhcpServerConfigIP4)) {
+                                        logger.debug("updating DhcpServerConfigIP4 for {}", interfaceName);
                                         configurationChanged = true;
                                         if (!modifiedInterfaceNames.contains(interfaceName)) {
                                             modifiedInterfaceNames.add(interfaceName);
                                         }
+                                    } else {
+                                        logger.debug("not updating DhcpServerConfigIP4 for {} because it is unchanged",
+                                                interfaceName);
                                     }
-                                    if (netConfig instanceof WifiConfig && !hadWifiConfig) {
-                                        logger.debug("adding new WifiConfig to existing config for {}", interfaceName);
-                                        newNetConfigs.add(netConfig);
-                                        configurationChanged = true;
-                                        if (!modifiedInterfaceNames.contains(interfaceName)) {
-                                            modifiedInterfaceNames.add(interfaceName);
-                                        }
+                                }
+                            } else if (netConfig instanceof FirewallAutoNatConfig) {
+                                if (natConfig == null) {
```

Comment:
Is this check needed?
if natConfig is null, I believe, it will not be an instance of FirewallAutoNatConfig, right?

Reply:
@MMaiero I believe this is needed. This code gets executed when NAT is disabled for interface in the 'DHCP & NAT' configuration tab. In this case netConfig is instanceof FirewallAutoNatConfig and natConfig is null. I verified it for both eth0 and wlan0.